core/src/main/java/com/amplifyframework/auth/options/AuthSignUpOptions.java

```diff
@@ -80,47 +75,73 @@ public boolean equals(Object obj) {
             return false;
         } else {
             AuthSignUpOptions authSignUpOptions = (AuthSignUpOptions) obj;
-            return ObjectsCompat.equals(getValidationData(), authSignUpOptions.getValidationData()) &&
-                    ObjectsCompat.equals(getUserAttributes(), authSignUpOptions.getUserAttributes());
+            return ObjectsCompat.equals(getUserAttributes(), authSignUpOptions.getUserAttributes());
         }
     }
 
+    /**
+     * When overriding, be sure to include the parent properties in the output string.
+     * @return A string representation of the object
+     */
     @Override
     public String toString() {
         return "AuthSignUpOptions{" +
                 "userAttributes=" + userAttributes +
-                ", validationData=" + validationData +
                 '}';
     }
 
-    public static final class Builder {
-        private Map<String, String> userAttributes;
-        private Map<String, String> validationData;
+    public abstract static class Builder<T extends Builder<T>> {
+        private List<AuthUserAttribute> userAttributes;
```

Comment:
the list reference can be `final`, since you only do `addAll` on it, not re-assign the reference, beyond the constrcutor.

Reply:
Gotcha - good point